client/src/reducers/controls.ts

```diff
@@ -439,7 +451,33 @@ const Controls = (
         mountCapture: false,
       };
     }
-
+    /**************************
+         Uns/Spatial
+    **************************/
+    case "toggle image underlay": {
+      return {
+        ...state,
+        imageUnderlay: !state.imageUnderlay,
+      };
+    }
+    case "request uns metadata started":
+      return {
+        ...state,
+        loading: true,
+        error: null,
+      };
+    case "request uns metadata success":
+      return {
+        ...state,
+        loading: false,
+        unsMetadata: action.data,
```

Comment:
Should we do the following, so any subsequent uns metadata can be preserved instead of overwritten?

I'm assuming:

```ts
datasetUnsMetadataFetchAndLoad(dispatch, "spatial"); // returns an object with one key { spatial: { ... } }

datasetUnsMetadataFetchAndLoad(dispatch, "foo"); // returns an object with one key { foo: { ... } }
```

So the following reducer will yield state:

```ts
unsMetadata: {
  spatial: { ... },
  foo: { ... }
}
```

```suggestion
        unsMetadata: { 
          ...state.unsMetadata,  // copy the current unsMetadata object
          ...action.data, // only overwrite the key/value pair from `action.data`
        },
```

Reply:
yes good call